design/src/rfcs/rfc0022_error_context_and_compatibility.md

```diff
@@ -0,0 +1,370 @@
+RFC: Error Context and Compatibility
+====================================
+
+> Status: RFC
+>
+> Applies to: Generated clients and shared rust-runtime crates
+
+This RFC proposes a pattern for writing Rust errors to provide consistent
+error context AND forwards/backwards compatibility. The goal is to strike
+a balance between these four goals:
+
+1. Errors are forwards compatible, and changes to errors are backwards compatible
+2. Errors are ergonomic
+3. Error messages are easy to debug
+4. Errors adhere to Rust's `Error` trait
```

Comment:
did you mean:
```suggestion
4. Errors implement Rust's `Error` trait
```

Reply:
Implementing `Error` is a given. My intent here was to communicate that they should adhere to the design/standards/best practices established by the `Error` trait. For example, the optional `source()` function should be implemented where possible.